v2/blockstore/readonly_test.go

```diff
@@ -269,3 +271,46 @@ func TestReadOnlyErrorAfterClose(t *testing.T) {
 	// TODO: test that closing blocks if an AllKeysChan operation is
 	// in progress.
 }
+
+func TestNewReadOnly_WorksWithIoReader(t *testing.T) {
```

Comment:
it isn't for io.Reader though, right? an io.Reader won't be able to seek back - does this mean that an open call on an actual `io.Reader` stream will fail?

Reply:
Good point the test name is confusing; thanks for pointing that out

Reply:
P.S. `ReadOnly` blockstore only takes `io.ReaderAt`